app/src/main/kotlin/com/wire/android/ui/login/Login.kt

```diff
@@ -0,0 +1,204 @@
+        keyboardOptions = KeyboardOptions(
+            keyboardType = KeyboardType.Text,
+            imeAction = ImeAction.Next
+        ),
+        modifier = modifier,
+    )
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+private fun PasswordInput(modifier: Modifier, password: TextFieldValue, onPasswordChange: (TextFieldValue) -> Unit) {
+    val keyboardController = LocalSoftwareKeyboardController.current
+    var passwordVisibility by remember { mutableStateOf(false) }
+    WireTextField(
+        value = password,
+        onValueChange = onPasswordChange,
+        placeholderText = stringResource(R.string.login_password_placeholder),
+        labelText = stringResource(R.string.login_password_label),
+        keyboardOptions = KeyboardOptions(
+            keyboardType = KeyboardType.Password,
+            imeAction = ImeAction.Done
+        ),
+        keyboardActions = KeyboardActions(onDone = { keyboardController?.hide() }),
+        visualTransformation = if (passwordVisibility) VisualTransformation.None else PasswordVisualTransformation(),
+        trailingIcon = {
+            val image = if (passwordVisibility) Icons.Filled.Visibility else Icons.Filled.VisibilityOff
+            IconButton(onClick = { passwordVisibility = !passwordVisibility }) {
+                Icon(imageVector = image, "", modifier = Modifier.size(20.dp))
+            }
+        },
+        modifier = modifier,
+    )
+}
+
+@Composable
+private fun ForgotPasswordLabel(modifier: Modifier) {
+    Column(horizontalAlignment = Alignment.CenterHorizontally, modifier = modifier) {
+        val context = LocalContext.current
+        Text(
+            text = stringResource(R.string.login_forgot_password),
+            style = MaterialTheme.typography.body1.copy(
+                textDecoration = TextDecoration.Underline,
+                color = MaterialTheme.colors.primary
+            ),
+            textAlign = TextAlign.Center,
+            modifier = Modifier
+                .clickable(
+                    interactionSource = remember { MutableInteractionSource() },
+                    indication = null,
+                    onClick = { Toast.makeText(context, "Forgot password click 💥", Toast.LENGTH_SHORT).show() } //TODO
+                )
+        )
+    }
+}
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+private fun LoginButton(modifier: Modifier, email: String, password: String) {
+    var isLoading by remember { mutableStateOf(false) }
+    val interactionSource = remember { MutableInteractionSource() }
+    Column(modifier = modifier) {
+        val enabled = validInput(email, password) && !isLoading
+        Button(
+            interactionSource = interactionSource,
+            shape = RoundedCornerShape(16.dp),
+            elevation = ButtonDefaults.elevation(0.dp, 0.dp, 0.dp),
+            colors = AnimatedButtonColors(enabled = enabled),
+            onClick = { isLoading = true }, //TODO
```

Comment:
loading state should be handled by the viewModel

Reply:
Yes it can be in the ViewModel, current solution is temporary just to get the UI working, I assumed that this PR would only add UI, the logic and ViewModel would be added in the next